artcommon/artcommonlib/registry_config.py

```
@@ -0,0 +1,328 @@
+
+            self.logger.debug(f'Wrote {len(self.merged_auths)} total auth entries to {self.temp_auth_file}')
+            self.logger.info(f'Enabled registries: {list(self.merged_auths.keys())}')
+
+            # Verify writing was successful
+            with open(self.temp_auth_file, 'r') as f:
+                verify_config = json.load(f)
+                verify_auths = verify_config.get('auths', {})
+                if len(verify_auths) != len(self.merged_auths):
+                    raise RuntimeError(
+                        f'Verification failed: Expected {len(self.merged_auths)} entries, got {len(verify_auths)}'
+                    )
```

Ruff (0.14.10)

Abstract `raise` to an inner function

(TRY301)